lightningd/onchain_control.c

```diff
@@ -518,6 +518,43 @@ static u32 infinite_block_deadline(const struct chain_topology *topo)
 	return get_block_height(topo) + 300;
 }
 
+/**
+ * slow_sweep_dealine -- A smart(-ish) way of estimating sweep fees.
+ *
+ * We use the `close_blockheight` as the basis for the target. We then
+ * target to have the funds back at most 2 weeks after the close,
+ * linearly increasing the confirmation target until we get
+ * close. However, we also don't want to panic and set a target that
+ * is too close, in order not to waste too many funds on the sweep
+ * fees.
+ */
+static u32 slow_sweep_deadline(const struct chain_topology *topo, const struct channel *c)
+{
+	u32 deadline, height = get_block_height(topo);
+	u32 closeheight = *c->close_blockheight ? *c->close_blockheight : 0;
+	if (closeheight == 0)
+		return get_block_height(topo) + 300;
+
+	/* A two week deadline seems like a not too bad default. We
+	 * should likely tweak this to hit the desired cost vs. idle
+	 * time tradeoff. */
+	deadline = closeheight + 2016;
+
+	/* If we missed the deadline, don't panic, let's continue with
+	 * a somewhat reasonable target, and hope for the best. No
+	 * need to ramp up to confirmation target 1, we're supposed to
+	 * be slow. */
+	if (deadline < height + 12)
```

Comment:
should we make these limits configurable?

Reply:
I prefer to come up with reasonable values before deferring to users. Generally speaking more config options don't really make software more usable. So let's wait for users to ask for customization rather than predicting what they'll want.

Reply:
Agreed.  We need a much more elaborate system if we want users to have visibility into, and acceleration of, transactions we are proposing.

Reply:
Also, to do this properly requires modifications elsewhere.   We set this deadline once, and then aim for it (modulo restarts).  We should have a "never urgent" flag in the code which uses the deadline to enforce a 12 block minimum target?

Reply:
After our earlier chat regarding how we should express intents, I think we need a bit more machinery for this kind of thing to become expressible. Ideally we'd have some priority queue of operations we'd like to perform, associated with a value, steering what we then actually do.